os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1320,6 +1331,8 @@ int up_lowgetc(void)
  ****************************************************************************/
 int up_putc(int ch)
 {
+	/*check if there is space in fifo*/
+	while(!LOGUART_Ready());
```

Comment:
Hi Mr @sunghan-chang, there is busy wait in rtl8721csm too, LOGUART_PutChar_RAM.
This is only for lldbg where it puts data directly into tx fifo. This busy wait is to check if the fifo has available space for data. Without this, if the fifo is not clear fast enough, there might be data missing when using lldbg.

For printf/dbg it will not have this busy wait because in uart_xmitchars there is a checking for uart_txready to prevent fifo full, and data loss

Reply:
sorry for type, `not` was missing.
`We should NOT do`
You look you understand what I wanted to say.

In `up_putc` in rtl8721csm, there is no uart tx ready.
Inside `loguart_putchar_ram`, it checks `tx_timeout` and `not_full`. But it's the same in rtl8730e.

Why does rtl8730e need checking this again in up_putc?
Let me assume an example,
1. wait tx ready here in up_putc
2. get ready and call up_lowputc and call loguart_putchar
3. and then immediately get context swiched
4. other thread adds logs and causes no tx buffer available
5. and then come back to this thread
6. log_uart_putchar should wait no_full again

I think it's possible to meet. Then, do we have any meaning to wait tx ready at up_putc?

Reply:
In this PR, i've removed that check in loguart_putchar (ameba_loguart.c), this check was causing performance issue. Therefore in rtl8730e loguart_putchar, it will not check for fifo not full, it relays on uart_txready to check, however for lldbg it does not go through uart_txready path, it just call loguart_putchar, without the busy-wait, there is a chance of data loss.